testsuite/pytests/sli2py_stimulating/test_step_current_generator.py

```diff
@@ -0,0 +1,367 @@
+result against expectation for one application.
+"""
+
+import nest
+import numpy.testing as nptest
+import pandas as pd
+import pandas.testing as pdtest
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def reset():
+    nest.ResetKernel()
+
+
+def test_set_params_on_instance_and_model_and():
+    """
+    Test setting ``step_current_generator`` parameters on instance and model.
+    """
+
+    params = {"amplitude_times": [1.0, 2.5, 4.6], "amplitude_values": [100.0, -100.0, 200.0]}
+
+    # Set params on instance
+    scg1 = nest.Create("step_current_generator")
+    scg1.set(params)
+
+    # Set params on model
+    nest.SetDefaults("step_current_generator", params)
+    scg2 = nest.Create("step_current_generator")
+
+    # Verify that both ways to set params give same result
+    df_scg1 = pd.DataFrame.from_dict(scg1.get(params.keys()))
+    df_scg2 = pd.DataFrame.from_dict(scg2.get(params.keys()))
+    pdtest.assert_frame_equal(df_scg1, df_scg2)
+
+
+@pytest.mark.parametrize("params", [{"amplitude_times": [1.0, 2.0]}, {"amplitude_values": [1.0, 2.0]}])
+def test_set_amplitude_times_or_values_only_fails(params):
+    """
+    Ensure failure if setting only one of ``amplitude_times`` or ``amplitude_values``.
+    """
+
+    with pytest.raises(nest.kernel.NESTErrors.BadProperty):
+        nest.SetDefaults("step_current_generator", params)
+
+
+def test_set_different_sized_amplitude_times_and_values_fails():
+    """
+    Ensure failure if setting different sized ``amplitude_times`` and ``amplitude_values``.
+    """
+
+    params = {"amplitude_times": [1.0, 2.0], "amplitude_values": [1.0, 2.0, 3.0]}
+
+    with pytest.raises(nest.kernel.NESTErrors.BadProperty):
+        nest.SetDefaults("step_current_generator", params)
+
+
+def test_fail_if_amplitude_times_not_strictly_increases():
+    """
+    Ensure failure if ``amplitude_times`` is not strictly increasing.
+    """
+
+    params = {"amplitude_times": [1.0, 2.0, 2.0], "amplitude_values": [1.0, 2.0, 3.0]}
+
+    with pytest.raises(nest.kernel.NESTErrors.BadProperty):
+        nest.SetDefaults("step_current_generator", params)
+
+
+def test_step_current_generator_simulation(reference_recording):
+    """
+    Test
+    """
+
+    nrn = nest.Create(
+        "iaf_psc_alpha", params={"V_th": 1e10, "C_m": 1.0, "tau_m": 1.0, "E_L": 0.0, "V_reset": 0.0, "V_m": 0.0}
+    )
+    scg = nest.Create(
+        "step_current_generator",
+        params={
+            "start": 5.0,
+            "stop": 15.0,
+            "amplitude_times": [1.5, 5.5, 7.5, 10.0, 14.9, 15.0, 20.0],
+            "amplitude_values": [1.0, -2.0, 4.0, -8.0, 16.0, -32.0, 64.0],
+        },
+    )
+    vm = nest.Create("voltmeter", params={"interval": 0.1})
+
+    nest.Connect(scg, nrn)
+    nest.Connect(vm, nrn)
+
+    nest.Simulate(25.0)
+
+    actual_recording = vm.events["V_m"]
+
+    nptest.assert_almost_equal(actual_recording, reference_recording, decimal=5)
+
+
+@pytest.fixture(scope="module")
+def reference_recording():
```

Comment:
Unfortunately, the original test does not say where the reference comes from, but the solution may have been explicitly computed since we know the closed-form solution. So I'd call it
```suggestion
def expected_V_m():
```